kubernetes/resource_kubernetes_env.go

```diff
@@ -0,0 +1,358 @@
+package kubernetes
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-kubernetes/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sschema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+)
+
+// TODO:
+/*
+* add support for cronjobs
+* add tests
+ */
+
+func resourceKubernetesEnv() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesEnvCreate,
+		ReadContext:   resourceKubernetesEnvRead,
+		UpdateContext: resourceKubernetesEnvUpdate,
+		DeleteContext: resourceKubernetesEnvDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"metadata": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "The name of the resource.",
+							Required:    true,
+							ForceNew:    true,
+						},
+						"namespace": {
+							Type:        schema.TypeString,
+							Description: "The namespace of the resource.",
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+			"container": {
+				Type:        schema.TypeString,
+				Description: "",
+				Required:    true,
+			},
+			"api_version": {
+				Type:        schema.TypeString,
+				Description: "API Version of Field Manager",
+				Required:    true,
+			},
+			"kind": {
+				Type:        schema.TypeString,
+				Description: "Type of resource being used",
+				Required:    true,
+			},
+			"env": {
+				Type:        schema.TypeList,
+				Description: "Rule defining a set of permissions for the role",
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Name of the environment variable. Must be a C_IDENTIFIER",
+						},
+						"value": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: `Variable references $(VAR_NAME) are expanded using the previous defined environment variables in the container and any service environment variables. If a variable cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded, regardless of whether the variable exists or not. Defaults to "".`,
+						},
```

Comment:
We're going to have to add in `valueFrom` here, see the schema for container here: https://github.com/hashicorp/terraform-provider-kubernetes/blob/83c23ce21474fa594ddf313f18210862197ba688/kubernetes/schema_container.go#L285-L404